x-pack/plugin/core/src/main/java/org/elasticsearch/license/XPackLicenseState.java

```diff
@@ -880,4 +802,24 @@ public static boolean isPlatinumOrTrialOperationMode(final OperationMode operati
     public synchronized XPackLicenseState copyCurrentLicenseState() {
         return new XPackLicenseState(this);
     }
+
+    private synchronized boolean checkSecurityEnabled() {
+        return isSecurityEnabled(status.mode, isSecurityExplicitlyEnabled, isSecurityEnabled);
+    }
+
+    private synchronized boolean checkMinimumLicense(
```

Comment:
I agree that the name is not good. I am not happy with the usage of `check` since this name usually feels like a void method. How about `isAllowedByLicenseAndSecurity`? I'll also add javadoc. Thanks

Reply:
[Updated](https://github.com/elastic/elasticsearch/pull/51864/files#diff-75ffc27a1f769f6d57256b3a4f04babcR823)